Moesif.Middleware/Helpers/AppConfigHelper.cs

```diff
@@ -13,16 +13,29 @@ namespace Moesif.Middleware.Helpers
 {
     public class AppConfigHelper
     {
-        public static async Task<AppConfig> getConfig(MoesifApiClient client, AppConfig prevAppConfig, bool debug)
+        public static AppConfig deepCopy(AppConfig config)
+        {
+            var json = ApiHelper.JsonSerialize(config);
+            var appConfig = ApiHelper.JsonDeserialize<AppConfig>(json);
+            appConfig.etag = config.etag;
+            appConfig.lastUpdatedTime = config.lastUpdatedTime;
+            return appConfig;
+        }
+       
+        public static async Task updateConfig(MoesifApiClient client, AppConfig config, bool debug)
         {
-            AppConfig appConfig = prevAppConfig;
 
             try
             {
                 var appConfigResp = await client.Api.GetAppConfigAsync();
-                appConfig = ApiHelper.JsonDeserialize<AppConfig>(appConfigResp.Body);
-                appConfig.etag = appConfigResp.Headers.ToDictionary(k => k.Key.ToLower(), k => k.Value)["x-moesif-config-etag"];
-                appConfig.lastUpdatedTime = DateTime.UtcNow;
+                var etag = appConfigResp.Headers.ToDictionary(k => k.Key.ToLower(), k => k.Value)["x-moesif-config-etag"];
+                if (etag != config.etag)
```

Comment:
@qmjava How does it guarantee that an old config will not be used to override current config?